src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -2944,6 +2944,8 @@ DlLockDataType DoorLockServer::credentialTypeToLockDataType(DlCredentialType cre
     case DlCredentialType::kFace:
         // So far there's no distinct data type for face credentials
         return DlLockDataType::kUnspecified;
+    case DlCredentialType::kUnknownEnumValue:
```

Comment:
There are also some places in the door lock ([here](https://github.com/project-chip/connectedhomeip/blob/8e26081616d736cd0427bdf3846da4220d80ea0a/src/app/clusters/door-lock-server/door-lock-server.cpp#L290), [here](https://github.com/project-chip/connectedhomeip/blob/8e26081616d736cd0427bdf3846da4220d80ea0a/src/app/clusters/door-lock-server/door-lock-server.cpp#L300), [here](https://github.com/project-chip/connectedhomeip/blob/8e26081616d736cd0427bdf3846da4220d80ea0a/src/app/clusters/door-lock-server/door-lock-server.cpp#L606), [here](https://github.com/project-chip/connectedhomeip/blob/8e26081616d736cd0427bdf3846da4220d80ea0a/src/app/clusters/door-lock-server/door-lock-server.cpp#L615) and [here](https://github.com/project-chip/connectedhomeip/blob/8e26081616d736cd0427bdf3846da4220d80ea0a/src/app/clusters/door-lock-server/door-lock-server.cpp#L2988)) where we check the range of the enum value. Maybe we can now simplify the checks and just verify the value against corresponding `kUnknownEnumValue`?

Reply:
Yes, that is the idea.  Filed https://github.com/project-chip/connectedhomeip/issues/20936